core/src/main/java/org/apache/iceberg/rest/RESTClient.java

```diff
@@ -54,11 +54,19 @@ default <T extends RESTResponse> T delete(
 
   <T extends RESTResponse> T delete(
       String path,
-      Map<String, String> queryParams,
       Class<T> responseType,
       Map<String, String> headers,
       Consumer<ErrorResponse> errorHandler);
 
+  default <T extends RESTResponse> T delete(
+      String path,
+      Map<String, String> queryParams,
```

Comment:
I'm not sure this makes sense for a default implementation because we're dropping query params.

Reply:
see https://github.com/apache/iceberg/pull/6146#discussion_r1018458745

Reply:
@danielcweeks suggested to throw an exception if `queryParams` are used so that we don't just silently drop them